common/src/main/java/tachyon/security/authentication/AuthenticationUtils.java

```diff
@@ -102,7 +102,9 @@ public static TTransport getClientTransport(TachyonConf tachyonConf,
    * @return An unconnected socket
    */
   public static TSocket createTSocket(InetSocketAddress address) {
-    return new TSocket(NetworkAddressUtils.getFqdnHost(address), address.getPort());
+    // TODO: make the timeout configurable.
+    return new TSocket(NetworkAddressUtils.getFqdnHost(address), address.getPort(),
+        30 * Constants.SECOND_MS);
```

Comment:
I wanted the smallest change for the minor release, so are we ok with adding a new config parameter (maybe tachyon.master.rpc.timeout.ms)?